nuxeo-services/nuxeo-platform-web-common/src/main/resources/web/nuxeo.war/login.jsp

```diff
@@ -524,6 +524,16 @@ input:-webkit-autofill:focus {
 </div>
 
 <script type="text/javascript">
+  // Handle requested URL for Web UI
+  var webUIPrefix = 'ui'
+  var requestedURL = document.getElementById('requestedUrl').value;
+  if (!requestedURL || requestedURL.startsWith(webUIPrefix)) {
+    var indexOfHash = window.location.href.indexOf('#!');
+    if (indexOfHash > -1) {
+      document.getElementById('requestedUrl').value = webUIPrefix + '/' + window.location.href.substring(indexOfHash);
+    }
+  }
+
```

Comment:
From my understanding of this chunk; there is an issue with a requestedUrl containing a `#!`.

Maybe it could be interesting to handle it globally instead of hardcoding a `web-ui` resource path. I think the redirect after login must handle the `#!` in his URL.